### PR TITLE
[WIP]adding Object item in specific position

### PR DESCRIPTION
**Pull request by beyond2void**
still  in work. 
Just wrote the function to add an object item in specific position.
But don't know where to call it.

### CODE REVIEW

**Comment by Logende** [comment]
You could put your function into the helpers folder and call the file something like insertPropertyAtPosition.ts (lower case name).

The relevant part in the code that is executed when the user changes data via GuiEditor (including when he adds new data) is 
https://github.com/PaulBredl/config-assistant/blob/2adc2b4aafa5416040f60d9f6e58926016c11958/frontend/config-assistant/src/components/gui-editor/GuiEditorPanelJsonSchema.vue#L17.

Here you could check whether the data already has the property and is just overwriting the value, or whether the property is not yet existing. If it does not yet exist, you could call your function to insert the new property at the desired position.
Probably, it is even more suitable to change the code in the store directly: https://github.com/PaulBredl/config-assistant/blob/2adc2b4aafa5416040f60d9f6e58926016c11958/frontend/config-assistant/src/store/sessionStore.ts#L214

File: frontend/config-assistant/src/components/gui-editor/ObjectItemToSpecificPosition.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
+export type AnyObject = {[key: string]: any};
+export function objectItemToSpecificPosition(
```
Comment (PaulBredl) [inline frontend/config-assistant/src/components/gui-editor/ObjectItemToSpecificPosition.ts:2]: Doc comment would be nice

**Comment by PaulBredl** [comment]
> You could put your function into the helpers folder and call the file something like insertPropertyAtPosition.ts (lower case name).
> 
> The relevant part in the code that is executed when the user changes data via GuiEditor (including when he adds new data) is
> 
> https://github.com/PaulBredl/config-assistant/blob/2adc2b4aafa5416040f60d9f6e58926016c11958/frontend/config-assistant/src/components/gui-editor/GuiEditorPanelJsonSchema.vue#L17
> 
> .
> Here you could check whether the data already has the property and is just overwriting the value, or whether the property is not yet existing. If it does not yet exist, you could call your function to insert the new property at the desired position. Probably, it is even more suitable to change the code in the store directly:
> 
> https://github.com/PaulBredl/config-assistant/blob/2adc2b4aafa5416040f60d9f6e58926016c11958/frontend/config-assistant/src/store/sessionStore.ts#L214

The difficulty is to find out the desired position. If the property is already in the data, its position shouldn't change.
If not, we need to use the schema to find out the position. This isn't trivial.

**Comment by PaulBredl** [comment]
@beyond2void this is not what Felix meant. The function was fine in the extra file. He meant that you should use in the session store

**Comment by beyond2void** [comment]
> @beyond2void this is not what Felix meant. The function was fine in the extra file. He meant that you should use in the session store

Yes, I was just thinking if It's also fine to put it in the session store, then we don't need an extra file?

**Comment by PaulBredl** [comment]
> > @beyond2void this is not what Felix meant. The function was fine in the extra file. He meant that you should use in the session store
> 
> Yes, I was just thinking if It's also fine to put it in the session store, then we don't need an extra file?

I would put it back in a separate file

**Comment by PaulBredl** [comment]
We decided to only do when we have time. We will keep the branch

**Pull request closed without merging**